src/movies.js

```diff
@@ -1,20 +1,46 @@
 // The `movies` array from the file `src/data.js`.
-console.log('movies: ', movies);
+// console.log('movies: ', movies);
 
 
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
+function getAllDirectors(arr) {
+  const directors = arr.map(element => {
+    return element.director;
+  })
+  return directors;
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies() {}
+function howManyMovies(arr) {
+  const dramas = arr.filter(movie => {
+    return movie.director === 'Steven Spielberg' && movie.genre.includes('Drama');
+  })
+  return dramas.length;
+}
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage() {}
+function scoresAverage(arr) {
+  if (arr.length === 0) {
+    return 0;
+  } else{
+  const scores = arr.reduce(function(accumulator, currentValue) {
+    return accumulator + currentValue.score
+  }, 0);
```

Comment:
the one test, that is failing here, you could have solved with an if condition like this:
```
 const scores = arr.reduce(function(accumulator, currentValue) {
    if (currentValue.score){
      return accumulator + currentValue.score
    }
    else {
      return accumulator
    }
  }, 0);
```